src/components/Table/utils/TableState.ts

```diff
@@ -147,141 +138,56 @@ export class TableState {
 
   // Updates the list of rows and regenerates the collapsedRows property if needed.
   setRows(rows: GridDataRow<any>[]): void {
+    // Note that because of using `rows: observable.shallow` above, this is always
+    // false, and this logic runs on every render. We can eventually fix this, but it
+    // is convenient b/c it puts no-longer-kept rows back into the right spot in their
+    // parents.
     if (rows !== this.rows) {
```

Comment:
**question:** Do you mean to say "this is always true" (`rows !== this.rows`)? That they'll never be equal due to the `observable.shallow`.